#### Se carga el data set 'iris-data.csv' y se observa su estructura.

In [1]:
import numpy as np
import pandas as pd

In [2]:
!curl https://raw.githubusercontent.com/plotly/datasets/master/iris-data.csv -o iris-data.csv

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  4757  100  4757    0     0  12616      0 --:--:-- --:--:-- --:--:-- 12651


In [3]:
iris_df = pd.read_csv('iris-data.csv')

In [4]:
print(iris_df.shape)

(150, 5)


In [6]:
print(iris_df[:5])
# Idealmente se realiza con iris_df.head()

   sepal length  sepal width  petal length  petal width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa


#### Se añaden lineas de código adicionales para realizar una descripción un poco más profunda del data set.

In [8]:
iris_df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal length  150 non-null    float64
 1   sepal width   150 non-null    float64
 2   petal length  150 non-null    float64
 3   petal width   150 non-null    float64
 4   class         150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [13]:
iris_df.isnull().sum()

sepal length    0
sepal width     0
petal length    0
petal width     0
class           0
dtype: int64

In [12]:
print(iris_df['class'].unique())

['Iris-setosa' 'Iris-versicolor' 'Iris-virginica']


#### Observamos que el data set se compone de cuatro variables numéricas flotantes: "sepal length" (longitud del sépalo), "sepal width" (ancho del sépalo), "petal length" (longitud del pétalo), "petal width" (ancho del pétalo); y una variable categórica de 3 categorías ('Iris setosa', 'Iris versicolor' e 'Iris virginica', que corresponden a distintos tipos de flores). 

#### Dentro del data set hay 150 filas o casos observados y no se observan valores nulos. 